modules/files.nix

```
@@ -343,8 +343,13 @@ in
           # We therefore simply log the conflict and otherwise ignore it, mainly
           # to make the `files-target-config` test work as expected.
           if [[ -e "$realOut/$relTarget" ]]; then
-            echo "File conflict for file '$relTarget'" >&2
-            return
+            if [[ -d "$realOut/$relTarget" && $(realpath "$realOut/$relTarget") == $realOut && $recursive ]]; then
+                # exception: allow a "collision" for a recureively linked home *directory*
```

```suggestion
            if [[ $recursive && -d "$realOut/$relTarget" && -d "$source" ]]; then
              # exception: allow a "collision" for a recursively linked *directory*
```

I don't think there's any reason to make a special case out of `target = "."`. Recursively-linked directories should work everywhere. Just test whether source and target are both directories so we know `lndir` will work.